run/cardano2john.py

```diff
@@ -0,0 +1,89 @@
+#!/usr/bin/env python3
+
+#
+# Export Cardano's legacy secret.key to John The Ripper format
+#
+# This software is Copyright (c) 2022, Pal Dorogi (ilap) <pal dot dorogi at gmail.com>
+# and it is hereby released to the general public under the following terms:
+#
+# Redistribution and use in source and binary forms, with or without
+# modification, are permitted.
+#
+# Requireements: pip3 at OS level
+#
+# Installation:
+# $ pip3 install argparse binascii cbor2
+#
+
+import sys
+import os
+
+import argparse
+import binascii
+from cbor2 import load
+
+
+def esk2john(user_key):
+    return "$cardano$1$" + binascii.hexlify(user_key[0]).decode("ascii")
+
+
+def export_to_john(filename):
+
+    try:
+        with open(filename, "rb") as fp:
+            # cbor[0]: _usVssMaybe, not interesting
+            # cbor[1]: _usPrimKey,  not interesting
+            # cbor[2]: _usKeys, interested only the esk part, atm, [[esk, passwordHash],...,]
+            # cbor[3]: _usWalletSet, interested if there is any, but only in its [0]
+            # i.e.: _wusRootKey which is an esk
+            cbor = load(fp)
+
+            # Go through all user's keys a.k.a _usKeys and the the Wallet set.
+            keys = cbor[2] + cbor[3]
+            keys_len = len(keys)
+
+            if keys_len == 0:
+                print("No keys found in the keystore file.")
+            else:
+                out_file = "cardano.esks"
+                try:
+                    with open(out_file, "w") as fp:
+                        print(
+                            "Exporting",
+                            keys_len,
+                            "keys to the %s file%s" % (out_file, os.linesep),
+                        )
+                        for line in map(esk2john, keys):
+                            fp.write(line + os.linesep)
+
+                except:
+                    sys.stderr.write("Cannot write file %s%s" % (out_file, os.linesep))
+                    sys.exit(1)
+    except:
+        sys.stderr.write("Wrong secret key file or path: %s%s" % (filename, os.linesep))
+        sys.exit(1)
+
+
+if __name__ == "__main__":
+
+    if len(sys.argv) < 2:
+        sys.stderr.write(
+            "Usage: %s -f <Cardano's secret.key file (./secret.key)>%s"
```

Comment:
Looks unnecessarily complicated and inconsistent with other 2john scripts (although I think in some we do have that?)

Instead, we should just expect the filename (or multiple) on the command line, without requiring any option. So no need for `argparse` in this script yet. We can add that later if needed for something other than an input filename.

Reply:
Refactored